packages/lesswrong/server/callbacks/postCallbacks.ts

```diff
@@ -520,3 +525,63 @@ getCollectionHooks("Posts").updateAfter.add(async (post: DbPost, props: CreateCa
 
   return post;
 });
+
+/**
+ * Call the Type3 webhook to notify it of a new post. This will trigger audio to be pre-generated for the post.
+ */
+async function callType3Webhook(action: 'post_published' | 'post_edited', url: string) {
+  const clientId = type3ClientIdSetting.get();
+  const webhookSecret = type3WebhookSecretSetting.get();
+
+  if (!clientId || !webhookSecret) return;
+
+  const webhookUrl = 'https://api.type3.audio/webhooks/create/';
+  const data = {
+    client_id: clientId,
+    action: action,
+    url: url,
+    key: webhookSecret,
+  };
+
+  const res = await fetch(webhookUrl, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(data),
+  });
```

Comment:
I sent the request this generates to Peter from type 3 and he says it's right, currently this returns a 404 though so it hits the log below